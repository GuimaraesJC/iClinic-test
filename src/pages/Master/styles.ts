import styled from 'styled-components';

import Button from '../../components/Button';

import darthVader from '../../assets/images/darth-vader.png';
import lukeSkywalker from '../../assets/images/luke-skywalker.png';

type Props = {
  master: string
}

export const Container = styled.main<Props>`
  display: flex;
  flex-direction: column;
  background-color: ${props => props.master === 'Darth Vader' ? '#2b2b2b' : '#fbfe63'};
  height: 100vh;
  align-items: center;
  position: relative;
  padding: 0 1rem;

  @media only screen and (max-width: 800px) {
    & > :nth-child(1) { order: 1; }
    & > :nth-child(2) { order: 4; }
    & > :nth-child(3) { order: 2; }
    & > :nth-child(4) { order: 3; }
  }
`;

export const StyledButton = styled(Button)`
  margin-top: 50px;
`;

export const ImageContainer = styled.div<Props>`
  width: 380px;
  height: 380px;
  border-radius: 50%;
  ${props => props.master === 'Darth Vader' ? '#fff;' : '#2a2a2a'};
  background: url(${props => props.master === 'Darth Vader' ? darthVader : lukeSkywalker});
  margin-top: 91px;

  @media only screen and (max-width: 800px) {
    margin-top: 210px;
  }
`;

export const MasterInfoContainer = styled.div`
  display: flex;
  align-items: center;
  margin-top: 38px;

  @media only screen and (max-width: 800px) {
    flex-direction: column;
  }
`;

export const MasterInfoText = styled.p<Props>`
  font-size: 36px;
  line-height: 44px;
  color: ${props => props.master === 'Darth Vader' ? '#fff;' : '#2a2a2a'};
`;

export const MasterInfoName = styled.span<Props>`
  font-size: 36px;
  line-height: 44px;
  color: ${props => props.master === 'Darth Vader' ? '#fff;' : '#2a2a2a'};
  font-weight: bold;
`;

export const ReturnContainer = styled.div<Props>`
  display: flex;
  align-items: center;
  color: ${props => props.master === 'Darth Vader' ? '#fff;' : '#2a2a2a'};
  position: absolute;
  top: 45px;
  left: 52px;

  transition: filter 0.2s;

  img {
    width: 30px;
    height: 25px;
  }

  span {
    font-size: 18px;
    line-height: 22px;
    margin-left: 10px;
  }

  &:hover {
    filter: brightness(0.9);
  }
`;
