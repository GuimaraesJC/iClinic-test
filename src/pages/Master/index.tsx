import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

import Button from 'components/Button';

import { fetchMasters } from 'utils/api';

import * as S from './styles';

import arrowLeft from 'assets/icons/arrow-left.svg';


function Master() {
  const [masterName, setMasterName] = useState('');
  const [isLoading, setIsLoading] = useState(true);

  function handleFetch() {
    setIsLoading(true);

    (async () => {
      const masterName = await fetchMasters();

      setMasterName(masterName);
      setIsLoading(false);
    })();
  }

  function handleClick() {
    handleFetch();
  }

  useEffect(() => {
    handleFetch();
  }, []);

  return (
    <S.Container master={masterName}>
      <Link to="/">
        <S.ReturnContainer master={masterName}>
          <img src={arrowLeft} alt="Back" />
          <span>back</span>
        </S.ReturnContainer>
      </Link>

      <S.ButtonContainer>
        <Button
          label="choose your path again, Padawan"
          size="large"
          font="normal"
          master={masterName}
          disabled={isLoading}
          onClickFn={handleClick}
        />
      </S.ButtonContainer>
      <S.ImageContainer master={masterName} data-testid="img-container"></S.ImageContainer>

      <S.MasterInfoContainer>
        <S.MasterInfoText master={masterName}>Your master is&nbsp;</S.MasterInfoText>
        <S.MasterInfoName master={masterName}>{masterName}</S.MasterInfoName>
      </S.MasterInfoContainer>
    </S.Container>
  );
}

export default Master;
