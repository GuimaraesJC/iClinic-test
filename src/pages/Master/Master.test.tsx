import { render, screen } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import Master from '.';

describe('<Master />', () => {
  it('should have a button with label "choose your path again, Padawan"', () => {
    render(
      <BrowserRouter>
        <Master />
      </BrowserRouter>
    );

    const buttonElement = screen.getByText(/choose your path again, padawan/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Master />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
