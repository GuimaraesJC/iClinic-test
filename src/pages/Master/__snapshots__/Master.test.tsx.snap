// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Master /> should match the snapshot 1`] = `
.c3 {
  width: 347px;
  height: 56px;
  background: #2c97d1;
  border: none;
  border-radius: 10px;
  color: #fff;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fbfe63;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  padding: 0 1rem;
}

.c2 {
  margin-top: 175px;
}

.c4 {
  width: 380px;
  height: 380px;
  border-radius: 50%;
  background: url(luke-skywalker.png);
  margin-top: 91px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 38px;
}

.c6 {
  font-size: 36px;
  line-height: 44px;
  color: #2a2a2a;
}

.c7 {
  font-size: 36px;
  line-height: 44px;
  color: #2a2a2a;
  font-weight: bold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
  position: absolute;
  top: 45px;
  left: 52px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c1 img {
  width: 30px;
  height: 25px;
}

.c1 span {
  font-size: 18px;
  line-height: 22px;
  margin-left: 10px;
}

.c1:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

@media only screen and (max-width:800px) {
  .c0 > :nth-child(1) {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c0 > :nth-child(2) {
    -webkit-order: 4;
    -ms-flex-order: 4;
    order: 4;
  }

  .c0 > :nth-child(3) {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }

  .c0 > :nth-child(4) {
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media only screen and (max-width:800px) {
  .c2 {
    margin-top: 50px;
  }
}

@media only screen and (max-width:800px) {
  .c4 {
    margin-top: 210px;
  }
}

@media only screen and (max-width:800px) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <main
    class="c0"
  >
    <a
      href="/"
    >
      <div
        class="c1"
      >
        <img
          alt="Back"
          src="arrow-left.svg"
        />
        <span>
          back
        </span>
      </div>
    </a>
    <div
      class="c2"
    >
      <button
        class="c3"
        disabled=""
        label="choose your path again, Padawan"
      >
        choose your path again, Padawan
      </button>
    </div>
    <div
      class="c4"
      data-testid="img-container"
    />
    <div
      class="c5"
    >
      <p
        class="c6"
      >
        Your master isÂ 
      </p>
      <span
        class="c7"
      />
    </div>
  </main>
</div>
`;
